### ISSUE TITLE
new Hih Res Fix

### ISSUE THREAD

**Issue by Blitzarts** [TheLastBen/fast-stable-diffusion#1231]
Hi!, Is there anyone else having trouble generating images over 768 x 1024 px with the high res option enabled? since now there is not only a high res fix, but you can also apply an upscaler, but if you do not use any, the image generates duplicate bodies and errors... and yes, if you choose one, the images are generated with problems of rendering (not like 1 week ago), before the update.

In addition, now with this upscale it takes twice as long to render the final images :c

![image](https://user-images.githubusercontent.com/120237309/210317835-59fea8b1-6a34-4ebc-a31f-d50260ea0ee5.png)

**Comment by Quark999** [comment]
According to the Automatic1111 releases, the idea is that you specify your "First pass" resolution as the resolution, and then apply the upscaling factor for the Hires pass - i.e. 512x768 with an upscaler of 2 creates a 1024x1536 picture.

That did work for me, but I started having the same issue than you with duplicate bodies, and also irregular aspect ratios gave me tons of issue. Currently trying to see whether restarting the colab helps - I think there are some gremlins in this, either in Automatic1111 or the Colab, or both.

**Comment by TheLastBen** [comment]
@Quark999 The colab only follows the A1111 repo

**Comment by Quark999** [comment]
> @Quark999 The colab only follows the A1111 repo

I had some weird error messages after a while that I wasn't sure were connected and might have been Colab-related, but I agree, it's the Hires changes in Automatic1111 that are the issue.

I think the issue is indeed with the resolution I had selected with the 768 model, as someone just pointed out on Automatic1111 - I went with 512x768 and upscale 2, but anything under 768 (and whatever upscale) appears to be causing the issues, and I think it always was like that and intended.

**Comment by Blitzarts** [comment]
yes, exactly. The same thing happens to me, I preferred the previous highres fix ahha. hopefully they can fix it or completely separate the options to avoid these problems. I think I will keep working with the local version until there is a fix, because as you say, it not only does upscale but also draws clones, double bodies, etc.

**Comment by Quark999** [comment]
> yes, exactly. The same thing happens to me, I preferred the previous highres fix ahha. hopefully they can fix it or completely separate the options to avoid these problems. I think I will keep working with the local version until there is a fix, because as you say, it not only does upscale but also draws clones, double bodies, etc.

I had the same problems with anything that wasn't a 1x1 ratio ever since I switched to the 768 model though - didn't you? So the new hires fix hasn't made things worse for me, and a 1x1 picture still upscales nicely just like it always did.

**Comment by Blitzarts** [comment]
I will explore that possible solution, but it happens that I am using mixed models of version 1.5, I don't know if it will cause me any problem?

**Comment by Quark999** [comment]
> I will explore that possible solution, but it happens that I am using mixed models of version 1.5, I don't know if it will cause me any problem?

I see - I think at least my issues are specifically caused by the 2.1 768 models, as the original issues were down to the Hires fix assuming a 512x512 "first pass", which doesn't work with the 768 models. So we might be talking about different things here - I did notice that I am having duplicated limbs even 1x1 aspect ratios now (although always with 2.1 / 768 models, which are the only ones that I am using) - perhaps the upscaler has something to do with it after all...

**Comment by Quark999** [comment]
> - I did notice that I am having duplicated limbs even 1x1 aspect ratios now (although always with 2.1 / 768 models, which are the only ones that I am using) - perhaps the upscaler has something to do with it after all...
No, turns out the GUI was still showing the previous run - 1x1 aspect ratio, 1.5 upscaler and 768x768 size on a 2.1 / 768 model does seem to work with the new hires fix. So it is indeed the uneven aspect ratio that appears to be broken in that scenario.
